roles/openshift_deploy_registry/tasks/main.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: Deploy registry
+  shell: osadm registry --create --credentials=/var/lib/openshift/openshift.local.certificates/openshift-registry/.kubeconfig --images='registry.access.redhat.com/openshift3_beta/ose-${component}:${version}'
+
+- name: Get registry deployment config
+  shell: osc get deploymentConfig/docker-registry -o json | sed -e '/"resourceVersion"/d'
+  register: output
+
+- name: Set registry node region
+  set_fact:
+    registry_fact: "{{ output.stdout | from_json | oo_set_resource_node({{ocom_infra_node_label}}) }}"
+
+- name: Write registry deployment config to temp file
+  copy:
+    content: "{{ registry_fact }}"
+    dest: /tmp/registry.json
+
+- name: Import new registry deployment config
+  shell: osc update deploymentConfig/docker-registry -f /tmp/registry.json
+
+- name: Remove registry deployment config temp file
+  file:
+    path: /tmp/registry.json
+    state: absent
```

Comment:
The problem we'll hit with this is that there will be different "supported" image streams for online, origin, and enterprise.  Not to mention origin/enterprise users will want some control over what imagestreams are available.

My initial take is that we create at least a datastructure to describe a minimally viable imagestream. We can then manipulate it in a jinja template and then pass the instantiated template to the create command.

Alternatively, we could have the base origin/online/enterprise files on disk, use those as defaults, and then allow the user to specify a local file to override the defaults.